### PR TITLE
Add out of the box modules page

### PR DESCRIPTION
**Pull request by ilayaperumalg**

### CODE REVIEW

File: spring-xd-out-of-the-box-modules.tex [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+Spring XD also supports developing a custom module for any usecase driven functionality.
+
+\subsection{Source modules}
+
+  These modules ingest data from outside into Spring XD which include, file, gemfire,
+http, jdbc, jms, kafka, mail, mqtt, rabbit, reactor-ip, reactor-syslog, sftp, syslog(tcp/udp),
+tail, tcp-client, tcp, twittersearch, twitterstreams.
+
+\subsection{Processor Modules}
+  These mdoules get the data out of source module or the previous processor module, computes
```
Comment (cppwfs) [inline spring-xd-out-of-the-box-modules.tex]: module is mis-spelled.

File: spring-xd-out-of-the-box-modules.tex [hunk 2]
```diff
@@ -0,0 +1,36 @@
+\section{Out of the box Modules}
+ Spring XD provides wide range of Streaming and batch Batch job modules out of the box which implement 
+some of the most commonly used streaming and batch job workflows. All the modules can readily be configured and deployed.
+Spring XD also supports developing a custom module for any usecase driven functionality.
+
+\subsection{Source modules}
+
+  These modules ingest data from outside into Spring XD which include, file, gemfire,
+http, jdbc, jms, kafka, mail, mqtt, rabbit, reactor-ip, reactor-syslog, sftp, syslog(tcp/udp),
+tail, tcp-client, tcp, twittersearch, twitterstreams.
```
Comment (cppwfs) [inline spring-xd-out-of-the-box-modules.tex]: twitterstream is singular

File: spring-xd-out-of-the-box-modules.tex (hunk 1 above)
Comment (cppwfs) [inline spring-xd-out-of-the-box-modules.tex]: probably use "modifies" instead of computes.

File: spring-xd-out-of-the-box-modules.tex [hunk 3]
```diff
@@ -0,0 +1,36 @@
+\section{Out of the box Modules}
+ Spring XD provides wide range of Streaming and batch Batch job modules out of the box which implement 
+some of the most commonly used streaming and batch job workflows. All the modules can readily be configured and deployed.
+Spring XD also supports developing a custom module for any usecase driven functionality.
+
+\subsection{Source modules}
+
+  These modules ingest data from outside into Spring XD which include, file, gemfire,
```
Comment (cppwfs) [inline spring-xd-out-of-the-box-modules.tex]: Replace "These modules" to "Source modules"

File: spring-xd-out-of-the-box-modules.tex [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+tail, tcp-client, tcp, twittersearch, twitterstreams.
+
+\subsection{Processor Modules}
+  These mdoules get the data out of source module or the previous processor module, computes
+the data and make the data available for the subsequent processor module or a sink module.
+These include aggregator, analytic-pmml, bridge, filter, http-client, json-to-tuple,
+object-to-json, script, shell command processor, splitter, transform.
+
+\subsection{Sink Modules}
+  These modules write the data into the final destination in the streaming application.
```
Comment (cppwfs) [inline spring-xd-out-of-the-box-modules.tex]: Replace "These modules" to "Sink modules"

File: spring-xd-out-of-the-box-modules.tex (hunk 1 above)
Comment (cppwfs) [inline spring-xd-out-of-the-box-modules.tex]: Replace "These modules" to "Processor
 modules"

Comment (ilayaperumalg) [inline spring-xd-out-of-the-box-modules.tex]: do you think "transforms" better? I like "modifies" as well.

**Comment by ilayaperumalg** [comment]
Updated the PR after fixing the review comments.

**Pull request merged**